### ISSUE TITLE
Custom validation messages?

### ISSUE THREAD

**Issue by uhoh-itsmaciek** [kiliman/remix-params-helper#5]
Thanks for writing this--this makes working with params in Remix much easier!

Have you considered adding customization to the parameter validation messages? The current messages are pretty inflexible. In particular, https://github.com/kiliman/remix-params-helper/blob/main/src/helper.ts#L67 leads to messages like "must be at least 8 characters for password" that sound pretty awkward. I could rewrite the message to improve this, like "8 is the required length for password", but that's not much better. Something like "${prop}${message}" would work better for my use case here, but I suspect that's not universally true (e.g., in some cases, if you're placing the message right next to the field that failed validation, you may want to not repeat the field name and have the error just pass through the zod validation error).

Maybe a callback that formats the Zod errors would work? Thoughts?

**Comment by kiliman** [comment]
Yeah, I should probably get rid of the error message handling and simply return what Zod uses. You can customize the Zod message in the schema.

```ts
const name = z.string({
  required_error: "Name is required",
  invalid_type_error: "Name must be a string",
});

z.string().min(5, { message: "Must be 5 or more characters long" });
z.string().max(5, { message: "Must be 5 or fewer characters long" });
z.string().length(5, { message: "Must be exactly 5 characters long" });
z.string().email({ message: "Invalid email address" });
z.string().url({ message: "Invalid url" });
z.string().uuid({ message: "Invalid UUID" });
```
https://github.com/colinhacks/zod#custom-error-messages

**Comment by uhoh-itsmaciek** [comment]
Fixed by 0035e1846adac57c0c4ab38400ee0f9132f263cc, right? Thanks!

**Comment by kiliman** [comment]
Yes sorry for not linking the issue.
